### PR TITLE
Ransomware landing page

### PR DESCRIPTION
**Pull request by VakarisZ**
# What does this PR do? 

Fixes parts of #1241

Add any further explanations here. 

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing? 
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Have you successfully tested your changes locally? Elaborate:
    > Tested by manual tests

## Screenshots:

landing_page

image

### CODE REVIEW

File: monkey/monkey_island/cc/ui/src/components/Main.tsx [hunk 1]
```diff
@@ -2,34 +2,65 @@ import React from 'react';
...
+import IslandHttpClient from "./IslandHttpClient";
 
-const reportZeroTrustRoute = '/report/zeroTrust';
-const islandModeRoute = '/api/island-mode'
+
+let notificationIcon = require('../images/notification-logo-512x512.png');
+
+const Routes = {
+  LandingPage: '/landing-page',
+  GettingStartedPage: '/',
```
Comment (ilija-lazoroski) [inline monkey/monkey_island/cc/ui/src/components/Main.tsx:38]: Shouldn't the landing page be `/` and the getting started page be `/getting-started`?

File: monkey/monkey_island/cc/ui/src/components/pages/LandingPage.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+function MonkeyInfo() {
+  return (
+    <>
+      <h4 className={'monkey-description-title'}>What is Infection Monkey?</h4>
+      <strong>Infection Monkey</strong> is an open-source security tool for testing a data center's resiliency to
+      perimeter
+      breaches and internal server infections. The Monkey uses various methods to propagate across a data center
+      and reports to this Monkey Island Command and Control server.
+    </>
```
Comment (ilija-lazoroski) [inline monkey/monkey_island/cc/ui/src/components/pages/LandingPage.tsx:71]: Isn't this duplicate to the Getting Started page ?

Comment (VakarisZ) [reply monkey/monkey_island/cc/ui/src/components/pages/LandingPage.tsx:71]: Yes, but this was a requirement we set in the issue

File: monkey/monkey_island/cc/ui/src/components/Main.tsx (hunk 1 above)
Comment (VakarisZ) [reply monkey/monkey_island/cc/ui/src/components/Main.tsx:39]: The `/landing-page` is only default as long as the mode is not chosen

File: monkey/monkey_island/cc/ui/src/components/layouts/DisabledSidebarLayoutComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+export const DisabledSidebarLayoutComponent = ({component: Component, ...rest}) => (
+  <Route {...rest} render={() => (
+    <Row>
+      <Col sm={3} md={3} lg={3} xl={2} className='sidebar'>
+        <SideNavComponent disabled={true} completedSteps={rest['completedSteps']}/>
+      </Col>
+      <Component {...rest} />
+    </Row>
+    )}/>
+)
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/ui/src/components/layouts/DisabledSidebarLayoutComponent.tsx:15]: This appears to be mostly duplicated from `StandardLayoutComponent`. Why not just pass a "sidebarDisabled" prop to StandardLayout component?

**Review by shreyamalviya — COMMENTED** [review]
Some questions (not all related to the PR)

File: monkey/monkey_island/cc/ui/src/components/Main.tsx [hunk 4]
```diff
@@ -2,34 +2,66 @@ import React from 'react';
...
 import 'react-toggle/style.css';
 import 'react-table/react-table.css';
-import notificationIcon from '../images/notification-logo-512x512.png';
 import {StandardLayoutComponent} from './layouts/StandardLayoutComponent';
 import LoadingScreen from './ui-components/LoadingScreen';
+import {DisabledSidebarLayoutComponent} from "./layouts/DisabledSidebarLayoutComponent";
+import {CompletedSteps} from "./side-menu/CompletedSteps";
+import Timeout = NodeJS.Timeout;
+import IslandHttpClient from "./IslandHttpClient";
+import _ from "lodash";
```
[the remark this answers is not in the document]
Comment (VakarisZ) [reply monkey/monkey_island/cc/ui/src/components/Main.tsx:32]: It's a common library for functional programming, has commonly used functions, like deep cloning objects.

File: monkey/monkey_island/cc/ui/src/components/SideNavComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+              <FontAwesomeIcon icon={faCheck} className='pull-right checkmark'/>
+              : ''}
+          </NavLink>
+        </li>
+        <li>
+          <NavLink to='/report/security'
+                   className={getNavLinkClass()}
+                   isActive={(_match, location) => {
+                     return (isReportRoute(location.pathname))
+                   }}>
```
[the remark this answers is not in the document]
Comment (VakarisZ) [reply monkey/monkey_island/cc/ui/src/components/SideNavComponent.tsx:59]: Set's the style of a report button in the sidenav  if the user is in the following url's (`/report/security`, `/report/attack`, etc.). This means that whichever report tab is chosen, report button is kept as active.
image

File: monkey/monkey_island/cc/ui/src/components/Main.tsx [hunk 6]
```diff
@@ -98,42 +145,26 @@ class AppComponent extends AuthComponent {
...
+  needsRedirectionToLandingPage = (route_path) => {
+    return (this.state.islandMode === null && route_path !== Routes.LandingPage)
+  }
+
+  needsRedirectionToGettingStarted = (route_path) => {
+    return route_path === Routes.LandingPage &&
+      this.state.islandMode !== null &&
+      this.state.islandMode !== undefined
+  }
+
```
[the remark this answers is not in the document]
Comment (VakarisZ) [reply monkey/monkey_island/cc/ui/src/components/Main.tsx:157]: You're right, the check for undefined is unnecessary since we set the default to `null`

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**